src/main/java/org/jabref/logic/importer/FulltextFetchers.java

```diff
@@ -36,26 +48,93 @@ public FulltextFetchers(ImportFormatPreferences importFormatPreferences) {
         Optional<DOI> doi = clonedEntry.getField(FieldName.DOI).flatMap(DOI::parse);
 
         if (!doi.isPresent()) {
+            findDoiForEntry(clonedEntry);
+        }
+
+        ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
We have the class `JabRefExecutorService` that manages executors. Reuse it here?